paderbox/io/hdf5.py

```diff
@@ -358,18 +360,37 @@ def ckeck_exists():
             # save strings, numpy.int64, and numpy.float64 types
             if isinstance(item, (np.int64, np.float64, np.float32,
                                  np.complex64, np.complex128,
-                                 str, complex, int, float)):
+                                 complex, int, float)):
                 ckeck_exists()
 
                 h5file[cur_path] = item
 
                 # NaN compares negative in Python.
                 # dataset.value has been deprecated. Use dataset[()] instead.
                 if not h5file[cur_path][()] == item and \
+                        not isinstance(item, str) and \
                         not np.isnan(item) and \
                         not np.isnan(h5file[cur_path][()]):
                     raise ValueError('The data representation in the HDF5 '
                                      'file does not match the original dict.')
+            elif isinstance(item, bytes):
+                ckeck_exists()
+
+                h5file[cur_path] = np.void(item)
+                if not h5file[cur_path][()] == np.void(item):
+                    raise ValueError('The data representation in the HDF5 '
+                                     'file does not match the original dict.')
+            elif isinstance(item, str):
+                ckeck_exists()
+                h5file[cur_path] = item
+
+                if LooseVersion(h5py.__version__) >= '3.0.0':
+                    test_item = item.encode('utf-8')
```

Comment:
Maybe we should add here a comment, that h5py changes in version 3 all strings to bytes and cannot distinguish between str and bytes. For bytes, we use the suggestion from the h5py documentation, how raw bytes streams can be stored.